pkg/harvester/validators/vm.js

```diff
@@ -87,7 +87,7 @@ export function vmDisks(spec, getters, errors, validatorArgs, displayKey, value)
         errors.push(getters['i18n/t']('harvester.validation.generic.tabError', { prefix, message }));
       }
 
-      if (typeValue?.spec?.resources?.requests?.storage && !/^([1-9][0-9]{0,8})[a-zA-Z]+$/.test(typeValue?.spec?.resources?.requests?.storage)) {
+      if (typeValue?.spec?.resources?.requests?.storage && !/^([0-9][0-9]{0,8})[a-zA-Z]+$/.test(typeValue?.spec?.resources?.requests?.storage)) {
```

Comment:
I also modify the regex : `/^([0-9][0-9]{0,8})[a-zA-Z]+$/` to allow user input 0 size. The API will return clear empty message for user.

<img width="1488" alt="Screenshot 2024-05-12 at 10 55 41 PM" src="https://github.com/harvester/dashboard/assets/5744158/04c736d8-6694-4174-9014-15cf069e0fcc">